org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/context/InjectBundleExtension.java

```diff
@@ -0,0 +1,150 @@
+
+	@Override
+	public void beforeAll(ExtensionContext extensionContext) throws Exception {
+		List<Field>
+
+		fields = findAnnotatedFields(extensionContext.getRequiredTestClass(), InjectBundle.class,
+			m -> Modifier.isStatic(m.getModifiers()));
+
+		fields.forEach(field -> {
+			assertFieldIsOfType(field, Bundle.class, InjectBundle.class, ExtensionConfigurationException::new);
+			InjectBundle injectBundle = field.getAnnotation(InjectBundle.class);
+			setField(field, null, getBundle(extensionContext, injectBundle));
+		});
+	}
+
+	@Override
+	public void beforeEach(ExtensionContext extensionContext) throws Exception {
+		for (Object instance : extensionContext.getRequiredTestInstances()
+			.getAllInstances()) {
+			final Class<?> testClass = instance.getClass();
+
+			List<Field> fields = findAnnotatedNonStaticFields(testClass, InjectBundle.class);
+
+			fields.forEach(field -> {
+				assertFieldIsOfType(field, Bundle.class, InjectBundle.class, ExtensionConfigurationException::new);
+				InjectBundle injectBundle = field.getAnnotation(InjectBundle.class);
+				setField(field, instance, getBundle(extensionContext, injectBundle));
+			});
+		}
+	}
+
+	/**
+	 * Resolve {@link Parameter} annotated with
+	 * {@link InjectBundleContext @BundleContextParameter} OR
+	 * {@link InjectInstallBundle @InstallBundleParameter} in the supplied
+	 * {@link ParameterContext}.
+	 */
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		Parameter parameter = parameterContext.getParameter();
+		Class<?> parameterType = parameter.getType();
+
+		if (parameterContext.isAnnotated(InjectBundle.class)) {
+			assertParameterIsOfType(parameterType, Bundle.class, InjectBundle.class, ParameterResolutionException::new);
+			InjectBundle injectBundle = parameter.getAnnotation(InjectBundle.class);
+			return getBundle(extensionContext, injectBundle);
+		}
+
+		throw new ExtensionConfigurationException("No parameter types known to BundleContextExtension were found");
+	}
+
+	/**
+	 * Determine if the {@link Parameter} in the supplied
+	 * {@link ParameterContext} is annotated with
+	 * {@link InjectBundleContext @BundleContextParameter} OR
+	 * {@link InjectInstallBundle @InstallBundleParameter}.
+	 */
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		boolean annotatedBundleParameter = parameterContext.isAnnotated(InjectBundle.class);
+
+		if (annotatedBundleParameter && (parameterContext.getDeclaringExecutable() instanceof Constructor)) {
+			throw new ParameterResolutionException(
+				"BundleContextExtension does not support parameter injection on constructors");
+		}
+		return annotatedBundleParameter;
+	}
+
+	private Bundle getBundle(ExtensionContext extensionContext, InjectBundle injectBundle) {
+
+		return installBundleOf(injectBundle, extensionContext);
+		// TODO:: ASK BJ extra store? Should work with InnstallBundle only.
```

Comment:
You make need to ask @rotty3000 here as I think he did InstallBundle.

Reply:
I am sure that we do not need to store it. the CloseableBundleContext will to the job.